### PR TITLE
fix(dependencies): Update PatternFly dependencies

### PR DESCRIPTION
**Pull request by mindreeper2420**
Update PatternFly to 3.30.1 and update the paths to account for the new structure and build requirements.

Fixes https://github.com/openshiftio/openshift.io/issues/1466

### CODE REVIEW

**Comment by fabric8cd** [comment]
@mindreeper2420 fabric8/fabric8-ui:SNAPSHOT-PR-2269-1 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2269-fabric8-ui.badger.fabric8.io

File: package.json [hunk 1]
```diff
@@ -176,6 +177,7 @@
     "extract-text-webpack-plugin": "3.0.2",
     "favicons-webpack-plugin": "0.0.7",
     "file-loader": "1.1.5",
+    "font-awesome": "4.7.0",
```
Comment (dlabrecq) [inline package.json]: Can bootstrap and font-awesome use the `^` char? Probably the last Bootstrap 3.x version, but the latest font-awesome is still v4.7.0

**Comment by mindreeper2420** [comment]
@dlabrecq Updated to use the `^` character.

**Comment by fabric8cd** [comment]
@mindreeper2420 fabric8/fabric8-ui:SNAPSHOT-PR-2269-2 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2269-fabric8-ui.badger.fabric8.io

File: config/webpack.common.js [hunk 2]
```diff
@@ -234,8 +234,10 @@ module.exports = function (options) {
                 loader: 'less-loader',
                 options: {
                   paths: [
-                    path.resolve(__dirname, "../node_modules/patternfly/src/less"),
-                    path.resolve(__dirname, "../node_modules/patternfly/node_modules")
+                    path.resolve(__dirname, "../node_modules/patternfly/dist/less"),
```
[the remark this answers is not in the document]
Comment (mindreeper2420) [reply config/webpack.common.js]: This is actually correct. PatternFly's module contains a `dist` folder where everything is located. There is no longer a `src` folder.

File: package.json [hunk 3]
```diff
@@ -165,6 +165,7 @@
     "assets-webpack-plugin": "3.5.1",
     "autoprefixer": "7.1.6",
     "awesome-typescript-loader": "3.2.3",
+    "bootstrap": "^3.3.7",
```
[the remark this answers is not in the document]
Comment (mindreeper2420) [reply package.json]: Removed, as I changed the references to just use the Bootstrap source in `patternfly/dist/dependencies/bootstrap`.

File: package.json [hunk 4]
```diff
@@ -176,6 +177,7 @@
     "extract-text-webpack-plugin": "3.0.2",
     "favicons-webpack-plugin": "0.0.7",
     "file-loader": "1.1.5",
+    "font-awesome": "^4.7.0",
```
[the remark this answers is not in the document]
Comment (mindreeper2420) [reply package.json]: Removed, as I changed the references to just use the Font Awesome source in `patternfly/dist/dependencies/font-awesome`.

File: package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (mindreeper2420) [reply package.json]: Removed, but any updates to PatternFly will now require that those be manually checked as well. The interlocked dependencies have been removed, which is why we wanted to stick with using 3.3.7 and anything above.

File: package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (mindreeper2420) [reply package.json]: Removed, but any updates to PatternFly will now require that those be manually checked as well. The interlocked dependencies have been removed, which is why we wanted to stick with using 4.7.0 and anything above.

Version 5.0 is RC5 at this point, but PatternFly does not have any plans to move to it.

**Comment by fabric8cd** [comment]
@mindreeper2420 fabric8/fabric8-ui:SNAPSHOT-PR-2269-3 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2269-fabric8-ui.badger.fabric8.io

File: config/webpack.common.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (joshuawilson) [reply config/webpack.common.js]: @dlabrecq why does PF-ng use a dist in the deployed npm module?

[the remark this answers is not in the document]
Comment (dlabrecq) [reply config/webpack.common.js]: PF-ng or PatternFly? These are PatternFly paths.

I don't know why the architect (at the time) chose to distribute PatternFly less files in the dist folder -- bootstrap; for example, doesn't distribute their less like that.

[the remark this answers is not in the document]
Comment (joshuawilson) [reply config/webpack.common.js]: ok, thanks

**Comment by fabric8cd** [comment]
@mindreeper2420 fabric8/fabric8-ui:SNAPSHOT-PR-2269-4 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2269-fabric8-ui.badger.fabric8.io

**Comment by fabric8cd** [comment]
@mindreeper2420 fabric8/fabric8-ui:SNAPSHOT-PR-2269-5 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2269-fabric8-ui.badger.fabric8.io

**Pull request merged**